bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/Journal.java

```diff
@@ -24,20 +24,24 @@
 import com.google.common.base.Stopwatch;
 import io.netty.buffer.ByteBuf;
 import io.netty.buffer.Unpooled;
+import io.netty.util.Recycler;
+import io.netty.util.Recycler.Handle;
+import io.netty.util.internal.RecyclableArrayList;
```

Comment:
It doesn't sound a good idea to use netty internal classes. can we avoid this? if we can avoid, I would prefer we have a similar util class. so we are not subject to netty changes.

Reply:
It's a little nasty, but any time we change netty versions we have a bunch of changes to make anyhow, right @merlimat